### PR TITLE
Add E2E test

### PR DESCRIPTION
**Pull request by jinSasaki**
- Fix bugs
- Add E2E test

@cre8ivejp 
Please let me know if there are any E2E test cases you would like to see. 🙏

### CODE REVIEW

File: Bucketeer2/Sources/Public/BKTClient.swift [hunk 1]
```diff
@@ -104,28 +71,56 @@ final class BKTClientImpl {
         }
     }
 }
-extension BKTClientImpl: BKTClient {
-    func stringVariation(featureId: String, defaultValue: String) -> String {
+
+extension BKTClient {
+    public static func initialize(config: BKTConfig, user: BKTUser, timeoutMillis: Int64 = 5000, completion: ((BKTError?) -> Void)?) {
```
Comment (cre8ivejp) [inline Bucketeer2/Sources/Public/BKTClient.swift:76]: Is it possible to implement it using async?

E.g.
```swift
initialize(config: BKTConfig, user: BKTUser, timeoutMillis: Int64 = 5000) async throws BKTError

// Await
let initError = try await BKTClient.initialize(config, user, timeoutMillis)
if initError != nil {
  ...
}

// Non-await
BKTClient.initialize(config, user)
```

**Review by cre8ivejp — CHANGES_REQUESTED** [review]

File: Bucketeer2Tests/BKTClientImplTests.swift [hunk 2]
```diff
@@ -1,21 +1,21 @@
 import XCTest
 @testable import Bucketeer2
 
-final class BKTClientImplTests: XCTestCase {
```
[the remark this answers is not in the document]
Comment (jinSasaki) [reply Bucketeer2Tests/BKTClientImplTests.swift:4]: Fixed https://github.com/ca-dp/bucketeer-ios-sdk/pull/88/commits/bb52db5460b6543bafe53eefa109d9cf45e58218

File: Bucketeer2/Sources/Public/BKTUser.swift [hunk 3]
```diff
@@ -4,33 +4,19 @@ public struct BKTUser: Equatable {
...
-            return self
-        }
-
-        public func build() throws -> BKTUser {
-            guard let id = self.id, !id.isEmpty else {
-                throw BKTError.illegalArgument(message: "The user id is required.")
-            }
-            return BKTUser(id: id, attributes: attributes)
-        }
+    public mutating func custom(attributes: [String: String]) {
```
[the remark this answers is not in the document]
Comment (jinSasaki) [reply Bucketeer2/Sources/Public/BKTUser.swift]: Fixed https://github.com/ca-dp/bucketeer-ios-sdk/pull/88/commits/b2787ded1b33614087e34fdef84a0e5df35fe615

File: Bucketeer2/Sources/Public/BKTClient.swift [hunk 4]
```diff
@@ -168,7 +163,7 @@ extension BKTClientImpl: BKTClient {
         }
     }
 
-    func evaluationDetails(featureId: String, completion: ((BKTEvaluation?) -> Void)?) {
+    public func evaluationDetails(featureId: String, completion: ((BKTEvaluation?) -> Void)?) {
```
[the remark this answers is not in the document]
Comment (jinSasaki) [reply Bucketeer2/Sources/Public/BKTClient.swift]: Fixed https://github.com/ca-dp/bucketeer-ios-sdk/pull/88/commits/10c68279e05b9c2a12896c4145b8b3c63bc156d3

File: Bucketeer2/Sources/Public/BKTClient.swift [hunk 5]
```diff
@@ -104,28 +71,56 @@ final class BKTClientImpl {
         }
     }
 }
-extension BKTClientImpl: BKTClient {
-    func stringVariation(featureId: String, defaultValue: String) -> String {
+
+extension BKTClient {
+    public static func initialize(config: BKTConfig, user: BKTUser, timeoutMillis: Int64 = 5000, completion: ((BKTError?) -> Void)?) {
+        guard BKTClient.default == nil else {
+            config.logger?.warn(message: "BKTClient is already initialized. not sure if initial fetch has been finished")
```
[the remark this answers is not in the document]
Comment (jinSasaki) [reply Bucketeer2/Sources/Public/BKTClient.swift]: Fixed https://github.com/ca-dp/bucketeer-ios-sdk/pull/88/commits/d044a3755ee73d42fbf5a97fcf1c68c5f8f18efd

File: Bucketeer2/Sources/Public/BKTClient.swift [hunk 6]
```diff
@@ -104,28 +71,56 @@ final class BKTClientImpl {
...
+            return
+        }
+        do {
+            let dispatchQueue = DispatchQueue(label: "io.bucketeer.taskQueue")
+            let dataModule = try DataModuleImpl(user: user.toUser(), config: config)
+            let client = BKTClient(dataModule: dataModule, dispatchQueue: dispatchQueue)
+            BKTClient.default = client
+            client.scheduleTasks()
+            client.refreshCache()
```
[the remark this answers is not in the document]
Comment (jinSasaki) [reply Bucketeer2/Sources/Public/BKTClient.swift]: Fixed https://github.com/ca-dp/bucketeer-ios-sdk/pull/88/commits/9dd84589018bbd73df023be7c2ce53c71d22c872

**Comment by jinSasaki** [comment]
Use feature flags of `sdke2e` project instead of the individual project. https://github.com/ca-dp/bucketeer-ios-sdk/pull/88/commits/a5edeff637f0554bf4446d52c9457f0a16a4ecc7

File: Bucketeer2/Sources/Public/BKTClient.swift (hunk 1 above)
Comment (cre8ivejp) [reply Bucketeer2/Sources/Public/BKTClient.swift:76]: Let's keep it as it is.
Thanks.

**Comment by jinSasaki** [comment]
@cre8ivejp 
The tests have been passed just now, so please review.
Sorry too late. 🙇

**Review by cre8ivejp — APPROVED** [review]
Excellent!
Thank you!

**Pull request merged**